### PR TITLE
transpiled using itself

### PR DESCRIPTION
**Pull request by AdamBurgess**
If I understood this right, plugin-babel was actually using traceur - this seems to add around 100kb to the build compared to babel! The minified build is now 7% smaller (with gzip that drops to 5%, but still significant).

This PR has the following changes:
* adds plugin-babel@0.0.5
* adds jspm.config.js to set transpiler (unless it can be set directly in package.json?)
* adds updated builds
* adds updated babel files (same babel semver though quite a lot of noise)
* changes the format of systemjs-babel-browser/node.js to CJS from AMD, and updates the plugin-babel package.json to explicitly set the format

The last change is a bit confusing - with AMD, the builds would fail to load. Even more so, you couldn't actually include the browser.js file in a bundle for fast reloading - it would fail with the same error: Unexpected anonymous AMD define. With CJS it all works fine...

this definitely needs testing

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cd build-babel
 jspm install --quick --log err
-jspm build systemjs-babel.js ../systemjs-babel-browser.js --format amd --skip-source-maps --minify --log ok
-jspm build systemjs-babel.js ../systemjs-babel-node.js --format amd --node --skip-source-maps --log ok
+jspm build systemjs-babel.js ../systemjs-babel-browser.js --format cjs --skip-source-maps --minify --log ok
+jspm build systemjs-babel.js ../systemjs-babel-node.js --format cjs --node --skip-source-maps --log ok
```
Comment (guybedford) [inline build.sh]: What was the reason for the change to `cjs` here? AMD as a format runs without source regexs so is in theory faster for larger files.

File: build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.5.2/lib/visit.js [hunk 2]
```diff
@@ -250,6 +250,6 @@ var awaitVisitor = {
     // Transforming `await x` to `yield regeneratorRuntime.awrap(x)`
     // causes the argument to be wrapped in such a way that the runtime
     // can distinguish between awaited and merely yielded values.
-    path.replaceWith(t.yieldExpression(t.callExpression(util.runtimeProperty("awrap", state.file.get("regeneratorIdentifier")), [argument]), false));
+    path.replaceWith(t.yieldExpression(t.callExpression(util.runtimeProperty("awrap"), [argument]), false));
   }
 };
```
Comment (guybedford) [inline build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.5.2/lib/visit.js]: This whole file is actually forked from the original, and needs to stay forked!

File: build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.5.2/lib/util.js [hunk 3]
```diff
@@ -20,8 +20,8 @@ var _babelTypes = require("babel-types");
 
 var t = _interopRequireWildcard(_babelTypes);
 
-function runtimeProperty(name, regeneratorIdentifier) {
-  return t.memberExpression(regeneratorIdentifier || t.identifier("regeneratorRuntime"), t.identifier(name), false);
+function runtimeProperty(name) {
+  return t.memberExpression(t.identifier("regeneratorRuntime"), t.identifier(name), false);
 }
 
 function isReference(path) {
```
Comment (guybedford) [inline build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.5.2/lib/util.js]: And this one too.

File: babel-helpers/slicedToArray.js [hunk 4]
```diff
@@ -1,35 +1,37 @@
...
+  return function (arr, i) {
+    if (Array.isArray(arr)) {
+      return arr;
+    } else if (Symbol.iterator in Object(arr)) {
+      return sliceIterator(arr, i);
+    } else {
+      throw new TypeError("Invalid attempt to destructure non-iterable instance");
+    }
+  };
+}();
```
Comment (guybedford) [inline babel-helpers/slicedToArray.js]: All these helper files are manually fixed actually to deal with Babel bug - http://phabricator.babeljs.io/T7136.

File: build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.5.2/lib/visit.js (hunk 2 above)
Comment (AdamBurgess) [inline build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.5.2/lib/visit.js]: had no idea, those files were replaced automatically by the build scripts ... I thought it was just a new version from babel so I kept them in.

**Comment by guybedford** [comment]
I've checked in the jspm.config.js and jspm.browser.js files to help clarify this.

File: build-babel/jspm.config.js [hunk 5]
```diff
@@ -0,0 +1,624 @@
+SystemJS.config({
+  packageConfigPaths: [
+    "npm:@*/*.json",
+    "npm:*.json",
+    "github:*/*.json"
+  ],
+  transpiler: "plugin-babel",
```
Comment (guybedford) [inline build-babel/jspm.config.js]: Huh, you're completely right here!

**Comment by guybedford** [comment]
It would be good to look into the AMD bug in the output, especially if that's a builder bug. I'm happy to work against a breaking PR.

**Comment by AdamBurgess** [comment]
The AMD bug is easily reproduceable, just create a bundle for plugin-babel (current version) and try including it. `jspm bundle plugin-babel babel-bundle.js` `<script src="babel-bundle.js"></script>`

**Comment by guybedford** [comment]
Thanks, unfortunately I just don't have the time to look into this too deeply right now. I'll aim to fix the AMD define bug though by letting the last define matched be the renamed one not the first.

Tracking this "feature" in https://github.com/systemjs/plugin-babel/issues/21.

Note that the only ES6 module file is the systemjs-babel.js file itself so I'm not sure how you found such a large benefit here.

**Pull request closed without merging**